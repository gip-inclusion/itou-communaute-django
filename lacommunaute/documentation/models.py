from django.conf import settings
from django.contrib.contenttypes.fields import GenericRelation
from django.db import models
from django.urls import reverse
from taggit.managers import TaggableManager

from lacommunaute.documentation.abstract_models import AbstractPublication
from lacommunaute.forum_upvote.models import UpVote
from lacommunaute.partner.models import Partner


class Category(AbstractPublication):
    class Meta:
        verbose_name = "Catégorie"
        verbose_name_plural = "Catégories"
        ordering = ["created_at"]

    def __str__(self):
        return f"{self.name}"

    def get_absolute_url(self, with_fqdn=False):
        absolute_url = reverse("documentation:category_detail", kwargs={"slug": self.slug, "pk": self.pk})
        if with_fqdn:
            return f"{settings.COMMU_PROTOCOL}://{settings.COMMU_FQDN}{absolute_url}"
        return absolute_url

    def get_update_url(self):
        return reverse("documentation:category_update", kwargs={"pk": self.pk, "slug": self.slug})


class Document(AbstractPublication):
    category = models.ForeignKey(Category, on_delete=models.CASCADE, related_name="documents")
    partner = models.ForeignKey(Partner, on_delete=models.CASCADE, null=True, blank=True)
    upvotes = GenericRelation(UpVote, related_query_name="document")
    certified = models.BooleanField(default=False, verbose_name="Certifié par la communauté de l'inclusion")
    tags = TaggableManager()

    class Meta:
        verbose_name = "Document"
        verbose_name_plural = "Documents"
        ordering = ["-created_at"]

    def __str__(self):
        return f"{self.name}"

    def get_absolute_url(self, with_fqdn=False):
        absolute_url = reverse(
            "documentation:document_detail",
            kwargs={
                "category_pk": self.category.pk,
                "slug": self.slug,
                "pk": self.pk,
            },
        )
        if with_fqdn:
            return f"{settings.COMMU_PROTOCOL}://{settings.COMMU_FQDN}{absolute_url}"
        return absolute_url


# use AbstractDatedModel after ForumRanting migration
class DocumentRating(models.Model):
    created_at = models.DateTimeField(null=True, blank=True)
    updated_at = models.DateTimeField(null=True, blank=True)
    session_id = models.CharField(max_length=40)
    document = models.ForeignKey(Document, on_delete=models.CASCADE)
    rating = models.PositiveSmallIntegerField()
    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, null=True, blank=True)

    class Meta:
        verbose_name = "Notation d'un document"
        verbose_name_plural = "Notations des documents"
        ordering = ("-created_at",)
