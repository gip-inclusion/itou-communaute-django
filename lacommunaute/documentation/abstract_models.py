from django.conf import settings
from django.db import models
from django.utils.encoding import force_str
from django.utils.text import slugify
from django.utils.translation import gettext_lazy as _
from machina.models.fields import MarkupTextField
from storages.backends.s3boto3 import S3Boto3Storage

from lacommunaute.utils.validators import validate_image_size


class AbstractDatedModel(models.Model):
    created_at = models.DateTimeField(auto_now_add=True)
    updated_at = models.DateTimeField(auto_now=True)

    class Meta:
        abstract = True


class AbstractPublication(AbstractDatedModel):
    name = models.CharField(max_length=100, verbose_name=_("Name"))
    slug = models.SlugField(max_length=255, verbose_name=_("Slug"), unique=True)

    description = MarkupTextField(verbose_name=_("Description"), null=True, blank=True)
    short_description = models.CharField(max_length=400, verbose_name="Description courte (SEO)")
    image = models.ImageField(
        blank=True,
        null=True,
        storage=S3Boto3Storage(bucket_name=settings.AWS_STORAGE_BUCKET_NAME, file_overwrite=False),
        validators=[validate_image_size],
    )

    class Meta:
        abstract = True

    def save(self, *args, **kwargs):
        self.slug = slugify(force_str(self.name), allow_unicode=True)
        super().save(*args, **kwargs)
