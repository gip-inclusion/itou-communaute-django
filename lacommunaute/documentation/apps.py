from django.apps import AppConfig


class ForumUpVoteAppConfig(AppConfig):
    label = "documentation"
    name = "lacommunaute.documentation"
    verbose_name = "Documentation"
    verbose_name_plural = "Documentation"
