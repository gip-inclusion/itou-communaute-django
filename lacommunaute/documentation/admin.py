from django.contrib import admin

from lacommunaute.documentation.models import Category, Document, DocumentRating


class DocumentInlines(admin.TabularInline):
    model = Document
    extra = 0
    fields = ("name", "short_description")
    readonly_fields = ("name", "short_description")

    def has_delete_permission(self, request, obj=None):
        return False

    def has_add_permission(self, request, obj=None):
        return False


@admin.register(Category)
class CategoryAdmin(admin.ModelAdmin):
    list_display = ("name",)
    search_fields = ("name",)
    fields = ("name", "short_description", "description", "image")
    inlines = [DocumentInlines]


@admin.register(Document)
class DocumentAdmin(admin.ModelAdmin):
    list_display = ("name", "category")
    list_filter = ("category", "partner")
    search_fields = ("name",)
    fields = ("name", "short_description", "description", "image", "category", "partner", "certified", "tags")


@admin.register(DocumentRating)
class DocumentRatingAdmin(admin.ModelAdmin):
    list_display = ("document", "rating", "created")
    list_filter = ("document",)
    list_display_links = ("rating",)
    raw_id_fields = ("document", "user")
