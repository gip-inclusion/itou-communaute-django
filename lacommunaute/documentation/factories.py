import factory
from faker import Faker

from lacommunaute.documentation.models import Category, Document, DocumentRating


faker = Faker()


class AbstractDocumentationFactory(factory.django.DjangoModelFactory):
    name = factory.Faker("name")
    description = factory.Faker("sentence", nb_words=100)
    short_description = factory.Faker("sentence", nb_words=10)
    image = factory.django.ImageField(filename="banner.jpg")


class CategoryFactory(AbstractDocumentationFactory):
    class Meta:
        model = Category

    class Params:
        for_snapshot = factory.Trait(
            name="Test Category", description="Test description", short_description="Test short description"
        )


class DocumentFactory(AbstractDocumentationFactory):
    category = factory.SubFactory(CategoryFactory)

    class Meta:
        model = Document

    class Params:
        for_snapshot = factory.Trait(
            name="Test Document",
            description="Test description",
            short_description="Test short description",
            category=factory.SubFactory(CategoryFactory, for_snapshot=True),
        )

    @factory.post_generation
    def with_tags(self, create, extracted, **kwargs):
        if not create or not extracted:
            return

        if isinstance(extracted, list):
            for tag in extracted:
                self.tags.add(tag)


class DocumentRatingFactory(factory.django.DjangoModelFactory):
    document = factory.SubFactory(DocumentFactory)
    rating = factory.Faker("random_int", min=1, max=5)

    class Meta:
        model = DocumentRating
        skip_postgeneration_save = True

    @factory.post_generation
    def set_created(self, create, extracted, **kwargs):
        if extracted:
            self.created = extracted
            self.save()
