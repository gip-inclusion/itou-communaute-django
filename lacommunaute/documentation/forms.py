from django import forms
from django.conf import settings
from taggit.models import Tag

from lacommunaute.documentation.models import Category, Document
from lacommunaute.partner.models import Partner
from lacommunaute.utils.iframe import wrap_iframe_in_div_tag


class DocumentationFormMixin:
    name = forms.CharField(required=True, label="Titre")
    short_description = forms.CharField(
        widget=forms.Textarea(attrs={"rows": 3}),
        max_length=400,
        required=True,
        label="Sous-titre (400 caractères pour le SEO)",
    )
    description = forms.CharField(
        widget=forms.Textarea(attrs={"rows": 20}), required=True, label="Contenu (markdown autorisé)"
    )
    image = forms.ImageField(
        label="Banniere de couverture, format 1200 x 630 pixels recommandé",
        widget=forms.FileInput(attrs={"accept": settings.SUPPORTED_IMAGE_FILE_TYPES.keys()}),
    )

    def save(self, commit=True):
        instance = super().save(commit=False)
        instance.description = wrap_iframe_in_div_tag(self.cleaned_data.get("description"))

        if commit:
            instance.save()
        return instance


class CategoryForm(forms.ModelForm, DocumentationFormMixin):
    class Meta:
        model = Category
        fields = ["name", "short_description", "description", "image"]


class DocumentForm(forms.ModelForm, DocumentationFormMixin):
    certified = forms.BooleanField(required=False, label="Certifiée par la communauté de l'inclusion")
    partner = forms.ModelChoiceField(
        label="Sélectionner un partenaire",
        queryset=Partner.objects.all(),
        required=False,
    )
    category = forms.ModelChoiceField(
        label="Sélectionner une catégorie documentaire",
        queryset=Category.objects.all(),
        required=False,
    )
    tags = forms.ModelMultipleChoiceField(
        label="Sélectionner un ou plusieurs tags",
        queryset=Tag.objects.all(),
        widget=forms.CheckboxSelectMultiple,
        required=False,
    )
    new_tags = forms.CharField(required=False, label="Ajouter un tag ou plusieurs tags (séparés par des virgules)")

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        if self.instance.pk:
            self.fields["tags"].initial = self.instance.tags.all()

    def save(self, commit=True):
        instance = super().save(commit=False)

        if commit:
            instance.save()
            instance.tags.set(self.cleaned_data["tags"])
            (
                instance.tags.add(*[tag.strip() for tag in self.cleaned_data["new_tags"].split(",")])
                if self.cleaned_data.get("new_tags")
                else None
            )
        return instance

    class Meta:
        model = Document
        fields = ["name", "short_description", "description", "image", "certified", "partner", "category"]
