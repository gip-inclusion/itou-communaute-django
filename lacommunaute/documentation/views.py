from django.contrib.contenttypes.models import ContentType
from django.views.generic import DetailView, ListView
from taggit.models import Tag

from lacommunaute.documentation.models import Category, Document


class CategoryListView(ListView):
    model = Category
    template_name = "documentation/category_list.html"
    context_object_name = "categories"
    paginate_by = 20 * 3


class CategoryDetailView(DetailView):
    model = Category
    template_name = "documentation/category_detail.html"
    context_object_name = "category"

    def get_tags_of_documents(self):
        return Tag.objects.filter(
            taggit_taggeditem_items__content_type=ContentType.objects.get_for_model(Document),
            taggit_taggeditem_items__object_id__in=self.object.documents.all().values_list("id", flat=True),
        ).distinct()

    def get_queryset(self):
        return super().get_queryset().prefetch_related("documents__tags")

    def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        context["tags"] = self.get_tags_of_documents()
        context["active_tag_slug"] = self.request.GET.get("tag") or None
        return context
