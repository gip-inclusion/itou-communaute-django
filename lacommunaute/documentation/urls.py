from django.urls import path

from lacommunaute.documentation.views import CategoryDetailView, CategoryListView, DocumentDetailView


app_name = "documentation"


urlpatterns = [
    path("", CategoryListView.as_view(), name="category_list"),
    path("<str:slug>-<int:pk>/", CategoryDetailView.as_view(), name="category_detail"),
]
