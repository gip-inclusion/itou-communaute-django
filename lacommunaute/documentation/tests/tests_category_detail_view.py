import pytest  # noqa

from django.urls import reverse

from lacommunaute.documentation.factories import CategoryFactory, DocumentFactory
from lacommunaute.utils.testing import parse_response_to_soup


@pytest.fixture(name="category")
def fixture_category():
    return CategoryFactory(for_snapshot=True)


@pytest.fixture(name="url")
def fixture_url(category):
    return reverse("documentation:category_detail", kwargs={"pk": category.pk, "slug": category.slug})


@pytest.mark.parametrize(
    "active_tag, snapshot_name",
    [
        ("tag1", "category_detail_view_with_active_tag"),
        (None, "category_detail_view_without_active_tag"),
        ("unknowntag", "category_detail_view_with_unknown_tag"),
    ],
)
def test_category_detail_view_with_tagged_documents(client, db, url, category, active_tag, snapshot_name, snapshot):
    DocumentFactory(category=category, with_tags=["tag1", "tag2"], for_snapshot=True)
    if active_tag:
        url = f"{url}?tag={active_tag}"
    response = client.get(url)
    assert response.status_code == 200
    content = parse_response_to_soup(response, selector="main", replace_img_src=True, replace_in_href=[category])
    assert str(content) == snapshot(name=snapshot_name)
