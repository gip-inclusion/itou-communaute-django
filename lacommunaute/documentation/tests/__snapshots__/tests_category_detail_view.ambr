# serializer version: 1
# name: test_category_detail_view_with_tagged_documents[None-category_detail_view_without_active_tag][category_detail_view_without_active_tag]
  '''
  <main class="s-main" id="main" role="main">
              
                  
              
              
      
  <div class="container">
      <nav aria-label="Fil d'ariane" class="c-breadcrumb">
          <ol class="breadcrumb">
              <li class="breadcrumb-item">Retourner vers</li>
              
                  <li class="breadcrumb-item">
                      <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="documentation_breadcrumb" href="/documentation/">Documentation</a>
                  </li>
              
          </ol>
      </nav>
  </div>
  
  
              
                  
      
      <section class="s-title-01 mt-lg-5">
      <div class="s-title-01__container container">
          <div class="s-title-01__row row">
              <div class="s-title-01__col col-12">
                  <h1>Test Category</h1>
                  
                  <h2 class="mt-3">Test short description</h2>
              </div>
          </div>
      </div>
  </section>
  
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
  
      <div class="col-12 col-sm-auto mb-3">
          <div class="d-none d-md-block forum-image mt-3" loading="lazy">
              <img alt="Test Category" class="rounded img-fluid" src="[img src]"/>
          </div>
      </div>
  
  <div class="col-12">
      <article class="textarea_cms_md"><p>Test description</p></article>
  </div>
  </div>
              </div>
          </section>
      
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
                      <div class="s-section__col col-12">
                          <div class="c-box">
                              <div class="container-fluid d-flex justify-content-center mb-3">Afficher les fiches contenant l'étiquette</div>
                              <div class="container-fluid d-flex justify-content-center">
                                  
                                      
                                          <button aria-label="Filtrer par tag1" class="tag bg-info-lighter text-info matomo-event" data-matomo-action="filter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag1">tag1</button>
                                      
                                  
                                      
                                          <button aria-label="Filtrer par tag2" class="tag bg-info-lighter text-info matomo-event" data-matomo-action="filter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag2">tag2</button>
                                      
                                  
                              </div>
                          </div>
                      </div>
                  </div>
              </div>
          </section>
      
      <section class="s-section">
          <div class="s-section__container container">
              <div class="s-section__row row">
                  <div class="s-section__col col-12">
                      <div class="row mt-4" id="documents">
                          
                              <div class="col-12 col-md-4 mb-3 mb-md-5">
      <div class="card c-card has-one-link-inside h-100">
          
              <div class="card-header card-header-thumbnail rounded">
                  <img alt="Test Document" class="img-fitcover img-fluid" loading="lazy" src="[img src]"/>
              </div>
          
          <div class="card-body pb-0">
              <p class="h3 lh-base">Test Document</p>
              <div>
                  <span class="tag bg-info-lighter text-info">tag1</span><span class="tag bg-info-lighter text-info">tag2</span>
              </div>
              <div class="mt-3">Test short description</div>
          </div>
          <div class="card-footer text-end">
              
                  <a class="btn btn-sm btn-ico btn-link stretched-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="document" href="#">
                      
                      <span>Consulter la fiche</span>
                      <i class="ri-arrow-right-line ri-lg"></i>
                  </a>
              
          </div>
      </div>
  </div>
  
                          
                      </div>
                  </div>
              </div>
          </div>
      </section>
  
              
          </main>
  '''
# ---
# name: test_category_detail_view_with_tagged_documents[tag1-category_detail_view_with_active_tag][category_detail_view_with_active_tag]
  '''
  <main class="s-main" id="main" role="main">
              
                  
              
              
      
  <div class="container">
      <nav aria-label="Fil d'ariane" class="c-breadcrumb">
          <ol class="breadcrumb">
              <li class="breadcrumb-item">Retourner vers</li>
              
                  <li class="breadcrumb-item">
                      <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="documentation_breadcrumb" href="/documentation/">Documentation</a>
                  </li>
              
          </ol>
      </nav>
  </div>
  
  
              
                  
      
      <section class="s-title-01 mt-lg-5">
      <div class="s-title-01__container container">
          <div class="s-title-01__row row">
              <div class="s-title-01__col col-12">
                  <h1>Test Category</h1>
                  
                  <h2 class="mt-3">Test short description</h2>
              </div>
          </div>
      </div>
  </section>
  
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
  
      <div class="col-12 col-sm-auto mb-3">
          <div class="d-none d-md-block forum-image mt-3" loading="lazy">
              <img alt="Test Category" class="rounded img-fluid" src="[img src]"/>
          </div>
      </div>
  
  <div class="col-12">
      <article class="textarea_cms_md"><p>Test description</p></article>
  </div>
  </div>
              </div>
          </section>
      
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
                      <div class="s-section__col col-12">
                          <div class="c-box">
                              <div class="container-fluid d-flex justify-content-center mb-3">Afficher les fiches contenant l'étiquette</div>
                              <div class="container-fluid d-flex justify-content-center">
                                  
                                      
                                          <button aria-label="Supprimer ce filtre" class="tag bg-info text-white matomo-event" data-bs-placement="top" data-bs-title="Supprimer ce filtre" data-bs-toggle="tooltip" data-matomo-action="unfilter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag1">
                                              <i class="ri-close-fill ri-xs"></i>tag1
                                          </button>
                                      
                                  
                                      
                                          <button aria-label="Filtrer par tag2" class="tag bg-info-lighter text-info matomo-event" data-matomo-action="filter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag2">tag2</button>
                                      
                                  
                              </div>
                          </div>
                      </div>
                  </div>
              </div>
          </section>
      
      <section class="s-section">
          <div class="s-section__container container">
              <div class="s-section__row row">
                  <div class="s-section__col col-12">
                      <div class="row mt-4" id="documents">
                          
                              <div class="col-12 col-md-4 mb-3 mb-md-5">
      <div class="card c-card has-one-link-inside h-100">
          
              <div class="card-header card-header-thumbnail rounded">
                  <img alt="Test Document" class="img-fitcover img-fluid" loading="lazy" src="[img src]"/>
              </div>
          
          <div class="card-body pb-0">
              <p class="h3 lh-base">Test Document</p>
              <div>
                  <span class="tag bg-info-lighter text-info">tag1</span><span class="tag bg-info-lighter text-info">tag2</span>
              </div>
              <div class="mt-3">Test short description</div>
          </div>
          <div class="card-footer text-end">
              
                  <a class="btn btn-sm btn-ico btn-link stretched-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="document" href="#">
                      
                      <span>Consulter la fiche</span>
                      <i class="ri-arrow-right-line ri-lg"></i>
                  </a>
              
          </div>
      </div>
  </div>
  
                          
                      </div>
                  </div>
              </div>
          </div>
      </section>
  
              
          </main>
  '''
# ---
# name: test_category_detail_view_with_tagged_documents[unknowntag-category_detail_view_with_unknown_tag][category_detail_view_with_unknown_tag]
  '''
  <main class="s-main" id="main" role="main">
              
                  
              
              
      
  <div class="container">
      <nav aria-label="Fil d'ariane" class="c-breadcrumb">
          <ol class="breadcrumb">
              <li class="breadcrumb-item">Retourner vers</li>
              
                  <li class="breadcrumb-item">
                      <a class="matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="documentation_breadcrumb" href="/documentation/">Documentation</a>
                  </li>
              
          </ol>
      </nav>
  </div>
  
  
              
                  
      
      <section class="s-title-01 mt-lg-5">
      <div class="s-title-01__container container">
          <div class="s-title-01__row row">
              <div class="s-title-01__col col-12">
                  <h1>Test Category</h1>
                  
                  <h2 class="mt-3">Test short description</h2>
              </div>
          </div>
      </div>
  </section>
  
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
  
      <div class="col-12 col-sm-auto mb-3">
          <div class="d-none d-md-block forum-image mt-3" loading="lazy">
              <img alt="Test Category" class="rounded img-fluid" src="[img src]"/>
          </div>
      </div>
  
  <div class="col-12">
      <article class="textarea_cms_md"><p>Test description</p></article>
  </div>
  </div>
              </div>
          </section>
      
      
          <section class="s-section">
              <div class="s-section__container container">
                  <div class="s-section__row row">
                      <div class="s-section__col col-12">
                          <div class="c-box">
                              <div class="container-fluid d-flex justify-content-center mb-3">Afficher les fiches contenant l'étiquette</div>
                              <div class="container-fluid d-flex justify-content-center">
                                  
                                      
                                          <button aria-label="Filtrer par tag1" class="tag bg-info-lighter text-info matomo-event" data-matomo-action="filter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag1">tag1</button>
                                      
                                  
                                      
                                          <button aria-label="Filtrer par tag2" class="tag bg-info-lighter text-info matomo-event" data-matomo-action="filter" data-matomo-category="engagement" data-matomo-option="document" hx-get="#" hx-swap="outerHTML" hx-target="#documents_area" id="filterdocument-tag2">tag2</button>
                                      
                                  
                              </div>
                          </div>
                      </div>
                  </div>
              </div>
          </section>
      
      <section class="s-section">
          <div class="s-section__container container">
              <div class="s-section__row row">
                  <div class="s-section__col col-12">
                      <div class="row mt-4" id="documents">
                          
                              <div class="col-12 col-md-4 mb-3 mb-md-5">
      <div class="card c-card has-one-link-inside h-100">
          
              <div class="card-header card-header-thumbnail rounded">
                  <img alt="Test Document" class="img-fitcover img-fluid" loading="lazy" src="[img src]"/>
              </div>
          
          <div class="card-body pb-0">
              <p class="h3 lh-base">Test Document</p>
              <div>
                  <span class="tag bg-info-lighter text-info">tag1</span><span class="tag bg-info-lighter text-info">tag2</span>
              </div>
              <div class="mt-3">Test short description</div>
          </div>
          <div class="card-footer text-end">
              
                  <a class="btn btn-sm btn-ico btn-link stretched-link matomo-event" data-matomo-action="view" data-matomo-category="engagement" data-matomo-option="document" href="#">
                      
                      <span>Consulter la fiche</span>
                      <i class="ri-arrow-right-line ri-lg"></i>
                  </a>
              
          </div>
      </div>
  </div>
  
                          
                      </div>
                  </div>
              </div>
          </div>
      </section>
  
              
          </main>
  '''
# ---
